mmv1/products/gkebackup/api.yaml

```diff
@@ -0,0 +1,238 @@
+        properties:
+          - !ruby/object:Api::Type::String
+            name: cronSchedule
+            description: |
+              A standard cron string that defines a repeating schedule for
+              creating Backups via this BackupPlan.
+              If this is defined, then backupRetainDays must also be defined.
+          - !ruby/object:Api::Type::Boolean
+              name: paused
+              description: |
+                This flag denotes whether automatic Backup creation is paused for this BackupPlan.
+      - !ruby/object:Api::Type::String
+        name: etag
+        output: true
+        description: |
+          etag is used for optimistic concurrency control as a way to help prevent simultaneous
+          updates of a backup plan from overwriting each other. It is strongly suggested that
+          systems make use of the 'etag' in the read-modify-write cycle to perform BackupPlan updates
+          in order to avoid race conditions: An etag is returned in the response to backupPlans.get,
+          and systems are expected to put that etag in the request to backupPlans.patch or
+          backupPlans.delete to ensure that their change will be applied to the same version of the resource.
+      - !ruby/object:Api::Type::Boolean
+        name: deactivated
+        description: |
+         This flag indicates whether this BackupPlan has been deactivated.
+         Setting this field to True locks the BackupPlan such that no further updates will be allowed
+         (except deletes), including the deactivated field itself. It also prevents any new Backups
+         from being created via this BackupPlan (including scheduled Backups).
+      - !ruby/object:Api::Type::NestedObject
+        name: backupConfig
+        description: |
+          Defines the configuration of Backups created via this BackupPlan.
+        properties:
+          - !ruby/object:Api::Type::Boolean
+            name: includeVolumeData
+            description: |
+              This flag specifies whether volume data should be backed up when PVCs are
+              included in the scope of a Backup.
+          - !ruby/object:Api::Type::Boolean
+            name: includeSecrets
+            description: |
+              This flag specifies whether Kubernetes Secret resources should be included
+              when they fall into the scope of Backups.
+          - !ruby/object:Api::Type::NestedObject
+            name: encryptionKey
+            description: |
+              This defines a customer managed encryption key that will be used to encrypt the "config"
+              portion (the Kubernetes resources) of Backups created via this plan.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: gcpKmsEncryptionKey
+                description: |
+                  Google Cloud KMS encryption key. Format: projects/*/locations/*/keyRings/*/cryptoKeys/*
```

Comment:
Do we want to have at least one of these sub field be set during config?

Reply:
all three of these fields are [optional](https://cloud.google.com/kubernetes-engine/docs/add-on/backup-for-gke/how-to/backup-plan#create), encryption key defaults to empty and includeVolumeData and includeSecrets default to false. I have includeVolumeData and includeSecrets as default_from_api in terraform.yaml file.